keps/sig-storage/2639-secret-protection/README.md

```diff
@@ -0,0 +1,450 @@
+# KEP-2639: Secret Protection
```

Comment:
should this be retitled to secrets and configmaps?

Reply:
Certainly "protection of in-use resources" is a general concept. The only resource-specific functionality would be the detection of "use". So it may be better to roll this all into one.

Reply:
@johnbelamaric 

Can this KEP go ahead with only secret first and then consider including other resources later or creating another KEP?
My intention is as commented [here](https://github.com/kubernetes/enhancements/pull/2640#discussion_r631350050) and [here](https://github.com/kubernetes/enhancements/pull/2640#discussion_r631942704). I think that we might find other resource than Configmap and this kind of investigations won't be the reason to block this feature.

Reply:
Actually, pv-protection and pvc-protection were meant to be a part of generic ["storageobjectinuseprotection"](https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/admission/storage/storageobjectinuseprotection) feature but missing these resources from the scope. I think that we should prioritize the impact and implement them.

Reply:
It seems likely given the hour and the level of unresolved questions that this is not going to make it by the deadline. The workday is already over where Jordan lives, for example. So I suggest considering asking for an exception. Or, taking a step back and looking at how we can address this more comprehensively in 1.23. It's usually better not to rush these things, especially when they're fairly complex.

Reply:
@johnbelamaric 

Thank you for your comment. I will wait for @liggitt 's response and ask for exception if it can.
If there are any concerns rest other than this, please point them out.

Reply:
I'm still against adding ConfigMap to the scope. However, it is easier to revert than to add, so I'm adding it to the last commit to make this KEP progress.